chain/chain/src/flat_storage_creator.rs

```diff
@@ -349,9 +351,9 @@ impl FlatStorageShardCreator {
                     }
                 }
 
-                Ok(())
+                Ok(false)
```

Comment:
Maybe a simpler way is to just return `Ok(current_status == FlatStorageStateStatus::Ready` at the end of the function instead of inside each case.